op-node/node/node.go

```diff
@@ -588,7 +588,9 @@ func (n *OpNode) onEvent(ev event.Event) bool {
 }
 
 func (n *OpNode) PublishBlock(ctx context.Context, signedEnvelope *opsigner.SignedExecutionPayloadEnvelope) error {
-	n.apiEmitter.Emit(tracer.TracePublishBlockEvent{Envelope: signedEnvelope.Envelope})
+	if n.cfg.Tracer != nil {
```

Comment:
Not sure if we want to wrap with `if` checks, or just introduce an `NoopTracer` is no tracer is configured. At the moment we emit only 2 Trace events, so it doesn't matter much, but if the tracer evolves, I'd probably opt for a NoopTracer.